packages/core/EventManager.ts

```diff
@@ -102,12 +107,16 @@ export default class EventManager {
     results.push(...(await this.createAllCalendarEvents(evt)));
 
     const referencesToCreate = results.map((result) => {
+      let createdEventObj: NewCalendarEventType | VideoCallData | null = null;
+      if (typeof result?.createdEvent === "string") {
+        createdEventObj = JSON.parse(result?.createdEvent);
```

Comment:
@Udit-takkar can use `zod` to parse the actual result and keep the type safety on runtime?

Reply:
ok